File: internal/content/testdata/bench/go/implement-trie.go
type trieNode struct {
	children [26]*trieNode
	isEnd    bool
}

type trie struct {
	root *trieNode
}

func newTrie() *trie {
	return &trie{root: &trieNode{}}
}

func (t *trie) insert(word string) {
	node := t.root
	for _, c := range word {
		idx := c - 'a'
		if node.children[idx] == nil {
			node.children[idx] = &trieNode{}
		}
		node = node.children[idx]
	}
	node.isEnd = true
}

func (t *trie) search(word string) bool {
	node := t.root
	for _, c := range word {
		idx := c - 'a'
		if node.children[idx] == nil {
			return false
		}
		node = node.children[idx]
	}
	return node.isEnd
}

func (t *trie) startsWith(prefix string) bool {
	node := t.root
	for _, c := range prefix {
		idx := c - 'a'
		if node.children[idx] == nil {
			return false
		}
		node = node.children[idx]
	}
	return true
}

func trie_ops(operations [][]string) []interface{} {
	t := newTrie()
	out := make([]interface{}, 0, len(operations))
	for _, op := range operations {
		opName := op[0]
		arg := op[1]
		switch opName {
		case "insert":
			t.insert(arg)
			out = append(out, nil)
		case "search":
			out = append(out, t.search(arg))
		default: // startsWith
			out = append(out, t.startsWith(arg))
		}
	}
	return out
}

File: internal/content/testdata/bench/typescript/implement-trie.ts
function trie_ops(operations: any[][]): any[] {
    const root: any = {};

    function insert(word: string): null {
        let node = root;
        for (const c of word) {
            if (!(c in node)) node[c] = {};
            node = node[c];
        }
        node["$"] = true;
        return null;
    }

    function search(word: string): boolean {
        let node = root;
        for (const c of word) {
            if (!(c in node)) return false;
            node = node[c];
        }
        return "$" in node;
    }

    function starts_with(prefix: string): boolean {
        let node = root;
        for (const c of prefix) {
            if (!(c in node)) return false;
            node = node[c];
        }
        return true;
    }

    const out: any[] = [];
    for (const [op, arg] of operations) {
        if (op === "insert") {
            out.push(insert(arg));
        } else if (op === "search") {
            out.push(search(arg));
        } else {
            out.push(starts_with(arg));
        }
    }
    return out;
}
